src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -84,6 +101,17 @@ public int ExecuteAsync()
             }
         }
 
+        internal void OnTestSessionTimeout(object obj)
+        {
+            string message = String.Format(ClientResources.TestSessionTimeoutMessage, this.testSessionTimeout);
+            var testMessagePayload = new CommunicationObjectModel.TestMessagePayload { MessageLevel = TestMessageLevel.Error, Message = message };
+            var rawMessage = this.dataSerializer.SerializePayload(CommunicationObjectModel.MessageType.TestMessage, testMessagePayload);
+
+            this.HandleLogMessage(TestMessageLevel.Error, message);
+            this.HandleRawMessage(rawMessage);
+            this.CancelAsync();
```

Comment:
Do we need any verbose logs in the changes in current PR? E.g. cancel is called by test host cleanup takes time, how can user know?